src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1315,7 +1315,7 @@ protected void SaveBackupFiles(object state)
                     workspace.SaveAs(savePath, null, true);
                     workspace.FileName = oldFileName;
                     workspace.Name = oldName;
-                    backupFilesDict.Add(workspace.Guid, savePath);
+                    backupFilesDict[workspace.Guid] = savePath;
```

Comment:
From time to time it fails because such key is already added into the dictionary. It's not an issue for DynamoSandbox or DynamoStudio because UI thread "swallows" exceptions preventing from crash, but if there is no WPF UI this exception crashes the application